### ISSUE TITLE
Different batchsize settings lead to different results

### ISSUE THREAD

**Issue by guggugg** [etzinis/sudo_rm_rf#8]
Hello, thank you for sharing. I trained this code on the LibriMix dataset. When the batchsize=1, the loss can drop normally, but when the batchsize=4, the loss is always positive. What could be the problem, can you give me some help?

**Comment by etzinis** [comment]
Thanks for reaching out. What does it mean drops normally? With batch size of 1 the results should be worse since with batch size of 1 there is no augmentation but I do not understand the always positive part.

**Comment by guggugg** [comment]
I use the [trainer ](https://github.com/JusperLee/Dual-Path-RNN-Pytorch). You see, here is set of batchsize=1,
`21-11-20 09:37:29 [F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRF.py:77 - INFO ] Train Datasets Length: 57657, Val Datasets Length: 4479
[F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRF.py:88 - INFO ] Building the Trainer of SuDORMRF
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:27 - INFO ] Load Nvida GPU .....
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:32 - INFO ] Loading SuDORMRF parameters: 5.002 Mb
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:61 - INFO ] Gradient clipping by 5, default L2
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:109 - INFO ] Start Validation from epoch: 0, iter: 0
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:133 - INFO ] <epoch:0, iter:1000, lr:1.000e-03, loss:29.278>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:133 - INFO ] <epoch:0, iter:2000, lr:1.000e-03, loss:29.318>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:133 - INFO ] <epoch:0, iter:3000, lr:1.000e-03, loss:29.429>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:133 - INFO ] <epoch:0, iter:4000, lr:1.000e-03, loss:29.361>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:139 - INFO ] Finished *** <epoch:0, iter:4480, lr:1.000e-03, loss:29.389, Total time:2.899 min> 
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:150 - INFO ] Starting epoch from 0, loss = 29.3892
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:67 - INFO ] Start training from epoch: 1, iter: 0
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:1000, lr:1.000e-03, loss:-0.425>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:2000, lr:1.000e-03, loss:-0.592>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:3000, lr:1.000e-03, loss:-0.879>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:4000, lr:1.000e-03, loss:-1.075>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:5000, lr:1.000e-03, loss:-1.204>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:6000, lr:1.000e-03, loss:-1.310>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:7000, lr:1.000e-03, loss:-1.406>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:8000, lr:1.000e-03, loss:-1.483>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:9000, lr:1.000e-03, loss:-1.561>`
and here the batchsize=4,
` [F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRFB.py:71 - INFO ] Building the optimizer of SuDORMRF
[F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRFB.py:75 - INFO ] Building the dataloader of SuDORMRF
[F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRFB.py:77 - INFO ] Train Datasets Length: 14415, Val Datasets Length: 1120
[F:/ZXX/Sudo-rm-rf-Pytorch-x/train/train-SuDORMRFB.py:88 - INFO ] Building the Trainer of SuDORMRF
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:27 - INFO ] Load Nvida GPU .....
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:32 - INFO ] Loading SuDORMRF parameters: 4.951 Mb
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:61 - INFO ] Gradient clipping by 5, default L2
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:109 - INFO ] Start Validation from epoch: 0, iter: 0
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:133 - INFO ] <epoch:0, iter:1000, lr:1.000e-03, loss:40.186>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:139 - INFO ] Finished *** <epoch:0, iter:1121, lr:1.000e-03, loss:40.155, Total time:1.597 min> 
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:150 - INFO ] Starting epoch from 0, loss = 40.1552
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:67 - INFO ] Start training from epoch: 1, iter: 0
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:1000, lr:1.000e-03, loss:30.551>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:2000, lr:1.000e-03, loss:30.522>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:3000, lr:1.000e-03, loss:30.510>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:4000, lr:1.000e-03, loss:30.462>
[F:\ZXX\Sudo-rm-rf-Pytorch-x\trainer.py:99 - INFO ] <epoch:1, iter:5000, lr:1.000e-03, loss:30.416>`

**Comment by guggugg** [comment]
Thanks for replying to me. I use the trainer [https://github.com/JusperLee/Dual-Path-RNN-Pytorch], and you see, here is the result of batchsize=1
![image](https://user-images.githubusercontent.com/73999879/142960834-26e955b3-6958-407f-9025-6b9965f7675a.png)
and here is the batchsize=4,
![image](https://user-images.githubusercontent.com/73999879/142960879-e0ba1c67-6023-46ee-850a-130ec7464d84.png)
I don't know why...

**Comment by etzinis** [comment]
If you use other trainers from other repos I can't possibly know what goes wrong.
